vignettes/faq.Rmd

```diff
@@ -0,0 +1,183 @@
+---
+title: "FAQ"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{FAQ}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+# Running and deployment
+
+<details>
+<summary>How to run a Rhino application?</summary>
+
+You can run a Rhino application exactly the same, as a regular Shiny app:
+
+- using `shiny::runApp()`
+- using "Run app" button in RStudio
+
+</details>
+
+<details>
+<summary>How to deploy a Rhino application?</summary>
+
+The same as in the case of a regular Shiny app, e.g. you can use "Deploy" button in RStudio IDE.
+
+</details>
+
+# Differences between Rhino and vanilla Shiny
+
+<details>
+<summary>Where are `server.R` and `ui.R` files?</summary>
+
+Instead of `server.R` and `ui.R`, Rhino uses a single file `main.R`.
+It includes both the server and UI part of the application.
+The main difference is that both are already Shiny modules so you
+need to use namespace (`ns`) in the UI part.
+
+</details>
+
+<details>
+<summary>Where is `global.R` file?</summary>
+
+Rhino encourages you to work with encapsulated modules instead of using global objects,
+thus it does not include `global.R` file.
+Instead, you can add code outside the server and UI parts of a module - it will be available
+only in this particular one, e.g.:
+
+```r
+box::use(
+  dplyr[collect, tbl]
+  shiny[moduleServer, NS],
+  pool[dbPool],
+  RSQLite[SQLite],
+)
+
+# connection would usually go to `global.R`
+connection <- dbPool(
+  SQLite(),
+  dbname = "database.sqlite"
+)
```

Comment:
In this particular case I'd create a dedicated module for the connection and export it. This way it can be imported anywhere in the app where it's needed - which is probably what a person looking for `global.R` needs in such case. WDYT?

Also: consider adding a link to our [How-to: Use global variables](https://appsilon.github.io/rhino/articles/how-to/use-global-variables.html).

Reply:
I'm afraid I have to disagree. Such an approach would lead to multiple existing connection objects in a single application. Instead, the connection should be created once and passed as an argument into modules that require it.

Regarding the How-to guide - I am not a big fan of this particular one, since it shows some hacks that I would like to avoid.

Reply:
Are we thinking about something different? Box caches modules, so the connection will be created only the first time the module is loaded with `box::use()`.

Reply:
Ok, that is a good point, but I still wouldn't use `box` for passing the connection. Having it as a single object visible in the code as an argument seems to be simpler and easier to understand and track than relying on the `box` features that happen under the hood. Also, if you think about testing, it seems easier and more understandable to pass a mocked connection.
Adding to that, usually, in our day-to-day work we use connections wrapped in R6 class objects (to reference that: [shiny.managers](https://github.com/Appsilon/shiny.managers) contains snippets for using this approach). So there is another layer of "state" that is passed across the application. While it would be (probably) possible to utilize the `box` cache to keep this R6 object in the environment created by `box`, this seems to me as a solution that is much less understandable than a simple "pass as an argument".

Reply:
No pressure, I don't have a perfect solution for dealing with DB in Shiny. My concern here is less about the particular example and more about how it fits the question being asked.

We are asked "Where is `global.R`?". I guess it can arise from different needs:
1. I want to put my `library()` calls somewhere.
2. I want to define a helper function or a constant and have it available everywhere.
3. I want a global mutable variable (e.g. to store a dataset and update it as the application runs).
4. I want to create an object (e.g. connection or pool) and have it available everywhere.

I think it would be good if the answer addressed these various needs. My suggestion is to make the core of the answer this: You need to think a bit differently when working with `box` and typically `box::use()` and `@export` is what you need.

The answer now only addresses (4) with a specific recommendation (don't make it available everywhere, pass it around explicitly). I nether agree nor disagree (IMO "it depends" :wink:) and I think it could be improved, but I already find it helpful and valuable as is. So use your own judgement here and feel free to merge without changes :slightly_smiling_face:

Reply:
It definitely makes sense to improve this section with answers to the listed questions, thank you for pointing that out!